Normalizer/Flat/ReferenceDataNormalizer.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Pim\Bundle\CustomEntityBundle\Normalizer\Flat;
+
+use Akeneo\Component\Localization\Model\TranslatableInterface;
+use Akeneo\Component\Localization\Model\TranslationInterface;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Util\ClassUtils;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Pim\Bundle\CustomEntityBundle\Reflection\ClassMetadataRegistry;
+use Pim\Bundle\CustomEntityBundle\Reflection\ReflectionClassRegistry;
+use Pim\Component\ReferenceData\Model\ReferenceDataInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @author    Romain Monceau <romain@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ */
+class ReferenceDataNormalizer implements NormalizerInterface
+{
+    /** @var ReflectionClassRegistry */
+    protected $reflectionClassRegistry;
+
+    /** @var ClassMetadataRegistry */
+    protected $classMetadataRegistry;
+
+    /** @var PropertyAccessorInterface */
+    protected $propertyAccessor;
+
+    /** @var NormalizerInterface */
+    protected $transNormalizer;
+
+    /** @var array */
+    protected $skippedFields = [];
+
+    /**
+     * @param ReflectionClassRegistry $reflectionClassRegistry
+     * @param ClassMetadataRegistry $classMetadataRegistry
+     * @param PropertyAccessorInterface $propertyAccessor
+     * @param NormalizerInterface $transNormalizer
+     */
+    public function __construct(
+        ReflectionClassRegistry $reflectionClassRegistry,
+        ClassMetadataRegistry $classMetadataRegistry,
+        PropertyAccessorInterface $propertyAccessor,
+        NormalizerInterface $transNormalizer
+    ) {
+        $this->reflectionClassRegistry = $reflectionClassRegistry;
+        $this->classMetadataRegistry = $classMetadataRegistry;
+        $this->propertyAccessor = $propertyAccessor;
+        $this->transNormalizer = $transNormalizer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = []): array
```

Comment:
I would have help to rework this part. Let me know what seems the better for you

Reply:
Do you mean you would **need** help?